docs/memory.md

```diff
@@ -79,6 +86,47 @@ ThisFeature.test_adapter.on!(:flag_name, context: user) # with context
 ThisFeature.test_adapter.off!(:flag_name)               # without context
 ```
 
+### **#enable_treatment!**
+
+This method is useful when you need to enable a feature flag with a treatment (or multiple treatments), and not just `"on"` and `"off"`.
+
+Usage example of these:
+
+```ruby
+# If you have configured the in-memory adapter as the default
+ThisFeature.test_adapter.enable_treatment!(:flag_name, treatment: 'treatment_name', context: user)
+```
+#### This method requires 3 arguments:
+1. `flag_name`: String or Symbol (not a named argument)
+2. `treatment`: String
+3. `context`: User or Org object
+
+Per flag name, there can only be one treatment per `context_key` (the ID of object that is provided as `context`). So multiple calls with the same `context`, but different treatment names, will be overwritten.
+
+```ruby
+ThisFeature.test_adapter.enable_treatment!('flag_a', treatment: 'treatment_1', context: user1)
+ThisFeature.test_adapter.storage # => { 'flag_a' => { :treatment_contexts => { 'User1': 'treatment_1' } } }
+
+ThisFeature.test_adapter.enable_treatment!(:flag_a, treatment: 'treatment_2', context: user1)
+ThisFeature.test_adapter.storage # => { 'flag_a' => { :treatment_contexts => { 'User1': 'treatment_2' } } }
+```
+
+### **#treatment_value**
+
+You can retrieve the flag's treatment name for a specific context.
+
+Usage example of these:
+
+```ruby
+# If you have configured the in-memory adapter as the default
+ThisFeature.test_adapter.treatment_value(:flag_name, context: user)
+```
+
+#### This method requires 2 arguments:
+1. `flag_name`: String or Symbol (not a named argument)
+2. `context`: User or Org object
+
+When the Memory storage does not contain the given flag_name or if there is no provided `context`, `"control"` is returned. This is meant to mimic what SplitIO would return in the case of no configured treatment for the given `context`.
```

Comment:
See SplitIO's docs on their [`Control treatment`](https://help.split.io/hc/en-us/articles/360020528072-Control-treatment)

The Memory Adapter doesn't mimic SplitIO exactly. With SplitIO, depending on whether or not the flag has been configured or has been configured with treatments, you'll get either `"control"` or `"control_treatment"` respectively, as a response.

I didn't know if it was important to mirror this functionality exactly for our Memory Adapter. Because these methods, and the Memory Adapter itself, are meant for use in the test environment only. And of course, because SplitIO can change their output at my time.

Aside from that, there's other details I felt like a rabbit-hole discussion could've been started from. Such as returning `false` or `nil`, from a method that is meant to return 1 of potentially many values. Values that are not a simple on/off switch. Would `false` mean something different from `nil`, when there are options like `v1`, `v2`, `v3`, etc